engine/src/main/java/org/pentaho/di/repository/kdr/delegates/metastore/KettleDatabaseRepositoryMetaStore.java

```diff
@@ -0,0 +1,460 @@
+    }
+  }
+
+  @Override
+  public IMetaStoreElementType getElementTypeByName( String namespace, String elementTypeName ) throws MetaStoreException {
+    try {
+      LongObjectId namespaceId = delegate.getNamespaceId( namespace );
+      if ( namespaceId == null ) {
+        return null;
+      }
+      LongObjectId elementTypeId = delegate.getElementTypeId( namespaceId, elementTypeName );
+      if ( elementTypeId == null ) {
+        return null;
+      }
+      RowMetaAndData elementTypeRow = delegate.getElementType( elementTypeId );
+
+      return delegate.parseElementType( namespace, namespaceId, elementTypeRow );
+    } catch ( Exception e ) {
+      throw new MetaStoreException( "Unable to get element type with name '"
+        + elementTypeName + "' in namespace '" + namespace + "'", e );
+    }
+  }
+
+  @Override
+  public void updateElementType( String namespace, IMetaStoreElementType elementType ) throws MetaStoreException {
+    try {
+      ObjectId namespaceId = delegate.verifyNamespace( namespace );
+      String elementTypeId = elementType.getId();
+      if ( elementTypeId == null ) {
+        IMetaStoreElementType type = getElementTypeByName( namespace, elementType.getName() );
+        if ( type != null ) {
+          elementTypeId = type.getId();
+        }
+      }
+
+      if ( elementTypeId != null ) {
+        delegate.updateElementType(
+          namespaceId, new LongObjectId( new StringObjectId( elementType.getId() ) ), elementType );
+        repository.commit();
+      } else {
+        throw new MetaStoreException( "Unable to update element type: no id was provided and the name '"
+          + elementType.getName() + "' didn't match" );
+      }
+    } catch ( MetaStoreException e ) {
+      throw e;
+    } catch ( Exception e ) {
+      repository.rollback();
+      throw new MetaStoreException( "Unable to update element type", e );
+    }
+  }
+
+  @Override
+  public void deleteElementType( String namespace, IMetaStoreElementType elementType ) throws MetaStoreException,
+    MetaStoreDependenciesExistsException {
+    try {
+      Collection<RowMetaAndData> elementTypeRows =
+        delegate.getElements( new LongObjectId( new StringObjectId( elementType.getId() ) ) );
+      if ( !elementTypeRows.isEmpty() ) {
+        List<String> dependencies = new ArrayList<String>();
+        for ( RowMetaAndData elementTypeRow : elementTypeRows ) {
+          Long elementTypeId =
+            elementTypeRow.getInteger( KettleDatabaseRepository.FIELD_ELEMENT_TYPE_ID_ELEMENT_TYPE );
+          dependencies.add( Long.toString( elementTypeId ) );
+        }
+        throw new MetaStoreDependenciesExistsException( dependencies, "The namespace to delete, '"
+          + namespace + "' is not empty" );
+      }
+
+      delegate.deleteElementType( new LongObjectId( new StringObjectId( elementType.getId() ) ) );
+      repository.commit();
+    } catch ( MetaStoreDependenciesExistsException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)